pkg/cmd/repo/create/create.go

```diff
@@ -624,27 +628,22 @@ func isLocalRepo(repoPath string) (bool, error) {
 }
 
 // clone the checkout branch to specified path
-func localInit(io *iostreams.IOStreams, remoteURL, path, checkoutBranch string) error {
-	gitInit, err := git.GitCommand("init", path)
+func localInit(gitClient *git.Client, remoteURL, path, checkoutBranch string) error {
+	gitInit, err := gitClient.Command(context.Background(), "init", path)
 	if err != nil {
 		return err
 	}
-	isTTY := io.IsStdoutTTY()
-	if isTTY {
-		gitInit.Stdout = io.Out
-	}
-	gitInit.Stderr = io.ErrOut
 	err = gitInit.Run()
 	if err != nil {
 		return err
 	}
 
-	gitRemoteAdd, err := git.GitCommand("-C", path, "remote", "add", "origin", remoteURL)
+	gitClient.RepoDir = path
```

Comment:
This mutates the original gitClient instance, creating a potentially surprising side-effect of `localInit`. It's probably fine, but for temporarily creating a client that is scoped to a specific repo, perhaps `git.Client` should have a method to create a clone of itself with a modified `RepoDir`?

Reply:
Yeah I was thinking that we might need a method like that. I was struggling with a name for it though.

Reply:
Added `CloneClient` command to address this.